Respect UI theme in token properties management panel, and other tweaks

kwvanderlinde opened the pull request:
### Identify the Bug or Feature request

Fixes #5888 

### Description of the Change

Makes some changes to `TokenPropertiesManagementPanel` so that the theme is better respected:
1. No more custom grid lines, instead using the default colours and intercell spacing.
2. Header cells use a header cell renderer instead of a regular cell renderer (`DefaultTableCellRenderer`), so that they are rendered according to the theme. This requires decorating the default renderer (done via `TableCellRendererDecorator`) since Java does not expose the header cell renderer implementation.
3. No more custom cell colours, in particular alternating column colours have been removed.

Some other UI tweaks:
1. The one label for the list of token types now says "Token Types" instead of "Token Type", to avoid confusion with the label for settings of a particular token type.
2. The "Help" button is now toggles when the help text is shown, instead of being a plain button with no toggle state.
3. Remove the multi-line header text / HTML in favour of plain text. Although this expands the widths of the checkbox columns and the short name column, there is still room for everything, especially with …
4. Reorganized the panel.
    - The type list no longer spans the entire panel top to bottom, but is in the top row alongside the token type settings. To make this fit comfortably, I've replaced the textual buttons for the type list with iconified alternatives, similar to the _Tables_ window.
    - The properties table is now on its own row that can span the entire width of the panel, gaining much needed horizontal space.
    - The help text has been relocated to the bottom so that the other elements don't have to move around when the help text is shown or hidden.

The total size of the panel is essentially the same as before. I've tested it on resolutions as low as 800x600, and for most themes the panel fits comfortably without the dialog needing to scroll. For a theme like **Ahh (Very Large Print)**, some scrolling is need to access the right-most parts of the panel, but this was already the case.

### Possible Drawbacks

The table columns are now equally spaced. Since everything fits, I'm not too worried about it, but it also means that the checkbox columns and short name columns take up more space than they need to.

### Documentation Notes

Old panel (from Cyan Light theme on nightly-20251128):
<img width="783" height="743" alt="image" src="https://github.com/user-attachments/assets/d8afa51d-bf04-4283-9bd7-8cc75aadfcfb" />

New panel:
<img width="783" height="745" alt="image" src="https://github.com/user-attachments/assets/b5cdd55c-3457-4f32-9baf-0bcb4f78125a" />

### Release Notes

- Modified the *Token Properties* tab of the **Campaign Properties** dialog to better follow the theme and to make better use of the screen real estate.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/RPTools/maptool/5892)
<!-- Reviewable:end -->

kwvanderlinde committed:
Make token properties help button be togglable

Now the button shows as selected whenever the help text is visible, so the user has a bit more feedback regarding its
functionality.

kwvanderlinde committed:
Add ACTION_ACCEPT icon

This just a checkmark. For the classic theme, it is the same as the checkmark used for indicating the current
initiative. For the Rod Takehara theme, a new icon was added parallel to the `ACTION_DELETE` icon.

kwvanderlinde committed:
Add utility for customizing JTable header cells.

`JTable` header cells are difficult to customize without messing up the theme since the default (and themable)
implementation for header cells is not publicly accessible and cannot be instantiated. Using `DefaultTableCellRenderer`
would cause the header cell to be themed as a regular table cell, not a header cell.

The new `TableCellRendererDecorator` can wrap any `TableCellRenderer` (in particular, the header cell renderer from an
existing table) and will modify the resultsin of the `#getTableCellRendererComponent()` to achieve the desired
customization.

Only vertical and horizontal text alignment are current supported, and requires the assumption that the decorated
renderer produces `JLabel` instance (which is true in practice).

kwvanderlinde committed:
Rework how token properties JTextFields are made to look like labels

The text fields are no longer focusable, to ensure they never receive a cursor.

Instead of setting a transparent background, we now assign the background of the parent container. This has the same
visual effect, but avoids some strange and probably buggy behaviour where the text would brighten if the user dragged
their mouse over the text field.

The labels and text fields now fill their assigned regions and rely on text alignment (rather than component alignment)
to position the label text properly.

kwvanderlinde committed:
Pluralize "token type" to "token types" for the type list

This distinguishes it from the identically named "Token Type" panel that contains settings for the currently selected
token type.

kwvanderlinde committed:
Remove custom table styling for token properties panel

- Remove custom grid colours and intercell spacing, instead using the theme defaults.
- Remove HTML from the column headers, meaning we no longer support line breaks.
- Remove the calculations for how large each column should be. Instead, we now use equally spaced columns. This does
  have a downside that the checkbox columns are provided more real estate than they need, but it makes the headers for
  those columns fit nicely.
- Remove the custom header height, as it is no longer needed now that our header text is on a single line.

kwvanderlinde committed:
Reorganize the token properties management panel

The goal here is to obtain extra horizontal space for the table - particularly the header cells - to display
nicely. Instead of the token type list sitting to the left of all other components, spanning the height of the pane,
the panel is now separated into three independent rows:
1. The first one contains the token type list, and the basic settings for the selected token type.
2. The second row contains the properties table, taking advantage of the full width of the dialog.
3. The third row contains the help text, which is hidden by default.

FullBleed commented:
I like the improvements, but I have a lot of Token Types (43 and growing).  Is that Token Types window fixed height or does it size at all when you stretch that dialog?

kwvanderlinde commented:
> I like the improvements, but I have a lot of Token Types (43 and growing). Is that Token Types window fixed height or does it size at all when you stretch that dialog?

Wow, that certainly is a lot!

The types list _is_ fixed height, but it does scroll:
<img width="778" height="742" alt="image" src="https://github.com/user-attachments/assets/9ed54c8b-1323-43ad-973e-4fc1feb808c8" />
So I don't think it will be much of a hit to usability. Though I could see that it could get tedious if you're doing a lot of back and forth changing and testing a property that is "below the fold".

We could tweak the amount of vertical space allotted to each row - the properties tables probably doesn't need that much vertical space. At the same time, though, I imagine properties outnumber property types in most cases, so it is better to optimize for a larger properties table vs a larger property type list.

kwvanderlinde committed:
Move the two main rows into a split pane

This allows the user to adjust the spacing if they want to focus more on property types or properties proper. The
default split is 30/70, giving ever so slightly more vertical space to the property table.

To get the propertions to make sense, this required removing some of the hardcoded preferred sizes. Those were not
actually needed anymore, so it's good for them to be gone anyways. An extra VSpacer was required to get the properties
table to stretch but not its button panel.

kwvanderlinde commented:
@FullBleed I've added adjustability, and modified the default allotment of the rows. The top row now has a bit more space by default, enough to show 11 property types vs just 8. And the divider in the middle can be dragged to view more types or more properties as desired.

<img width="783" height="742" alt="image" src="https://github.com/user-attachments/assets/039bb564-92a5-48b2-bcfd-2dbb8db6cc11" />

FullBleed commented:
> @FullBleed I've added adjustability, and modified the default allotment of the rows. The top row now has a bit more space by default, enough to show 11 property types vs just 8. And the divider in the middle can be dragged to view more types or more properties as desired.

Perfect solution.

Thanks!
